TrackletAlgorithm/MatchCalculator.h

```diff
@@ -227,6 +227,22 @@ void readTable_Cuts(ap_uint<width> table[depth]){
 // MatchCalculator
 namespace MC {
   enum imc {UNDEF_ITC, A = 0, B = 1, C = 2, D = 3, E = 4, F = 5, G = 6, H = 7, I = 8, J = 9, K = 10, L = 11, M = 12, N = 13, O = 14};
+
+  template<TF::layerDisk layer> constexpr regionType ASRegion();
+  template<TF::layerDisk layer> constexpr regionType APRegion();
+  template<TF::layerDisk layer> constexpr regionType FMRegion();
+
+  template<> constexpr regionType ASRegion<TF::L3>() {
+    return BARRELPS;
+  }
+
+  template<> constexpr regionType APRegion<TF::L3>() {
+    return BARRELPS;
+  }
+
+  template<> constexpr regionType FMRegion<TF::L3>() {
+    return BARREL;
+  }
```

Comment:
I added these lines to make the templating easier, but I'm not sure if this is the right file to put them in. Should we put things like this in Constants.h, or should the generation scripts be able to calculate the template parameters?

Reply:
As we move to automatically generating the top functions, I think the simplest thing is to have the scripts calculate these parameters directly. So, e.g., instead of the scripts generating a call to `ASRegion<TF::L3>()`, they would just write `BARRELPS` where appropriate. I'll need to update generate_TC.py to be consistent with this approach, but I think it's the way we should go.

Reply:
Done.